File: argi-go-client/src/app/models/Data/MeishiData.ts
import { IKotobaCreationOrUpdateData, IKotobaData } from '../IKotoba';
import { IKyougaku } from '../IKyougaku';

export interface MeishiData extends IKotobaData, IKyougaku {
  joukenKei: string;
}

export interface MeishiCreationOrUpdateData
  extends IKotobaCreationOrUpdateData,
    IKyougaku {
  joukenKei: string;
}

File: ArgiGo/Model/ModelData/Meishi/MeishiData.cs
﻿using ArgiGo.Model.ModelData.Chapters;
using ArgiGo.Model.ModelData.Exam;
using ArgiGo.Model.ModelData.Examples;

namespace ArgiGo.Model.ModelData.Meishi
{
    public class MeishiData : IKotobaData
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string Kanji { get; set; }

        public IEnumerable<string> Translation { get; set; }

        public string? JoukenKei { get; set; }

        public IEnumerable<ExampleData>? Examples { get; set; }

        public IEnumerable<ExamData> Exams { get; set; }

        public IEnumerable<ChapterData> Chapters { get; set; }
    }
}
